scientific_library/tvb/simulator/plot/plotter3D.py

```diff
@@ -0,0 +1,109 @@
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+#
+#
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+from tvb.simulator.lab import *
+import ipywidgets as widgets
+from IPython.display import display
+
+class Plotter3D:
+    def __init__(self):
+        pass
+        # self.sens_meg = sensors.SensorsMEG.from_file()
+        # self.conn = connectivity.Connectivity.from_file()
+        # self.skin = surfaces.SkinAir.from_file()
+        # self.sens_eeg = sensors.SensorsEEG.from_file()
+        # self.ctx = cortex.Cortex.from_file()
+        # self.loc_conn = local_connectivity.LocalConnectivity(cutoff=20.0, surface=self.ctx.region_mapping_data.surface)
+
+    def display_source_sensor_geometry(self, meg_sensors = sensors.SensorsMEG.from_file(),
+                                        eeg_sensors = sensors.SensorsEEG.from_file(),
+                                        conn = connectivity.Connectivity.from_file(),
+                                        skin = surfaces.SkinAir.from_file()):
+        box_layout = widgets.Layout(border='solid 1px black', margin='3px 3px 3px 3px', padding='2px 2px 2px 2px')
+        
+        skin.configure()
+        
+        eeg_sensors.configure()
+
+        # CONTROLS
+        params = dict()
+        roi_checkbox = widgets.Checkbox(description="Show ROI Centers", value=False)
+        eeg_checkbox = widgets.Checkbox(description="Show EEG Sensors", value=False)
+        meg_checkbox = widgets.Checkbox(description="Show MEG Sensors", value=False)
+
+        control_box = widgets.HBox([roi_checkbox,eeg_checkbox,meg_checkbox], layout=box_layout)
+
+        params['ROI'] = roi_checkbox
+        params['EEG'] = eeg_checkbox
+        params['MEG'] = meg_checkbox        
+
+        def plot(**plot_params):
+            ax = plt.subplot(111, projection='3d')
+
+            # Plot boundary surface
+            x, y, z = skin.vertices.T
+            ax.plot_trisurf(x, y, z, triangles=skin.triangles, alpha=0.1, edgecolor='none')
+
+            if plot_params['ROI']:
+                # ROI centers as black circles
+                x, y, z = conn.centres.T
+                ax.plot(x, y, z, 'ko')
+            
+            if plot_params['EEG']:
+                # EEG sensors as blue x's
+                x, y, z = eeg_sensors.sensors_to_surface(skin).T
+                ax.plot(x, y, z, 'bx')
+
+            if plot_params['MEG']:
+                x, y, z = meg_sensors.locations.T
+                ax.plot(x, y, z, 'r+')
+        plt.figure()
+
+        out = widgets.interactive_output(plot, params)
+        display(control_box,out)
+
+    def display_surface_local_connectivity(self, ctx = cortex.Cortex.from_file()):
```

Comment:
could the local_connectivity also be given as a parameter here? And in case no local_connectivity is provided, we  compute the one from above by default